docs/src/advanced/2_custom_model.md

```diff
@@ -1,2 +1,151 @@
 
-# Custom models
+# Working with custom models
+
+It often happens that the intuitive representation of your data does not really
+match what is supported by the given COBRA package. COBREXA.jl attempts to
+avoid this problem by providing a flexible framework for containing any data
+structure that can, somehow, represent the constraint-based model.
+
+The task of having such a polymorphic model definition can be split into 2
+separate concerns:
+
+- How to allow the analysis functions to gather the required information from
+  any user-specified model data structure?
+- How to make the reconstruction functions (i.e., reaction or gene deletions)
+  work properly on any data structure?
+
+To solve the first concern, COBREXA.jl specifies a set of generic accessors
+that work over the abstract type [`MetabolicModel`](@ref). To use your data
+structure in a model, you just make it a subtype of [`MetabolicModel`](@ref)
+and overload the required accessors. The accessors are functions that extract
+some relevant information, such as [`stoichiometry`](@ref) and
+[`bounds`](@ref), returning a fixed simple data type that can be further used
+by COBREXA.  You may see a complete list of accessors
+[here](../functions#Base-Types).
+
+A good solution to the second concern is a slightly more involved, as writing
+generic data modifiers is notoriously hard. Wtill, there is support for easily
+making small changes to the model using the modifications system, with
+functions such as [`with_added_reactions`](@ref) and [`with_set_bound`](@ref).
+
+!!! tip "Notebook available"
+    A better example of using a custom model structure is available
+    [in a separate notebook](../notebooks/8_custom_model.md).
+
+## Writing the generic accessors
+
+Let's write a data structure that represents a very small model that contains N
+metabolites that are converted in circle by N reactions. This may be useful for
```

Comment:
that are converted through linear, coupled reactions, e.g. A -> B -> C -> A with N= 3.